### DISCUSSION TITLE
Custom Docker client in api.grpc

### DISCUSSION THREAD

**Discussion by GoktugOcal** [coreemu/core#838] [Q&A]
Hi everyone, I am working on a projec that I want to limit the cpu and memory resources of nodes. I am using emulator to create nodes with docker container and I have defined my own docker node classes that I can set CPU and MEMORY limits as option parameters.

In my project I am using a grpc channel for communication and I assume that using core.api.grpc would be more appropriate for my case ( I have some grpc channel closing issue for lower bandwidths ). Can I define custom docker nodes for grpc module as well? I could not find a way to do that but maybe I have missed something. Is there any opinion to do that? Thank you everyone :)

**Comment by bharnden** [comment]
There is not a current way to achieve what you want. In the next release, where you can start nodes by way of `docker compose` that would be one way to set the options you would like.
